internal/provider/search_attribute_resource.go

```diff
@@ -0,0 +1,292 @@
+
+	tflog.Warn(ctx, "Update operation called, but updates are not supported for this resource.")
+
+}
+
+// Delete removes a Temporal search attribute from both Temporal and the Terraform state.
+func (r *SearchAttributeResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data SearchAttributeResourceModel
+
+	client := operatorservice.NewOperatorServiceClient(r.client)
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete request
+	request := &operatorservice.RemoveSearchAttributesRequest{
+		Namespace:        data.Namespace.ValueString(),
+		SearchAttributes: []string{data.Name.ValueString()},
+	}
+
+	_, err := client.RemoveSearchAttributes(ctx, request)
+
+	if err != nil {
+		if _, ok := err.(*serviceerror.NotFound); ok {
+			resp.Diagnostics.AddError("Request error", "Search attribute not found: "+err.Error())
+			return
+		}
+		resp.Diagnostics.AddError("Request error", "Unable to delete search attribute "+err.Error())
+		return
+	}
+
+	tflog.Info(ctx, fmt.Sprintf("Successfully deleted search attribute: %s", data.Name.ValueString()))
+	tflog.Trace(ctx, "Deleted a resource")
+}
+
+// ImportState allows existing Temporal search attributes to be imported into the Terraform state.
+// Importing system attributes is not supported
+func (r *SearchAttributeResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+
+	// Expected request ID format is a string 'namespace:search_attribute_name'
+	// Ex: 'default:CustomBool'
+
+	parts := strings.Split(req.ID, ":")
+	if len(parts) < 2 {
+		resp.Diagnostics.AddError("Invalid ID format", "Expected 'namespace:name'.")
+
+		return
+	}
+
+	namespace := parts[0]
+	name := parts[1]
+
+	// Fetch the search attribute details
+	client := operatorservice.NewOperatorServiceClient(r.client)
+	attrRequest := &operatorservice.ListSearchAttributesRequest{
+		Namespace: namespace,
+	}
+
+	attributes, err := client.ListSearchAttributes(ctx, attrRequest)
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read search attribute info, got error: %s", err))
+		return
+	}
+
+	// Find the specific attribute and get its type
+	var attributeType enums.IndexedValueType
+	var found bool
+
+	if attributeType, found = attributes.GetCustomAttributes()[name]; !found {
+		resp.Diagnostics.AddError("Not Found", fmt.Sprintf("Custom Search Attribute '%s' not found in namespace '%s'", name, namespace))
+		return
+	}
+
+	// Set the ID and other attributes needed for managing the resource
+	diags := resp.State.Set(ctx, &SearchAttributeResourceModel{
+		Name:      types.StringValue(name),
+		Namespace: types.StringValue(namespace),
+		Type:      types.StringValue(attributeType.String()),
+	})
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
```

Comment:
Logging the error here will be helpful for debugging.

Reply:
If there is an error and resp.Diagnostics.HasError() is true, the error will be automatically reported to the terraform runtime when the function returns. In most cases we catch errors and log them using resp.Diagnostics.AddError(), but here we are just checking if errors were generated. That's why we only need to ensure early return from the function if there were any issues